htdocs/fourn/commande/dispatch.php

```diff
@@ -1131,8 +1131,29 @@
 				print '<td>'.dol_print_date($db->jdate($objp->date_delivery), 'day').'</td>';
 
 				if (!empty($conf->productbatch->enabled)) {
-					print '<td class="dispatch_batch_number">'.$objp->batch.'</td>';
-					print '<td class="dispatch_dluo">'.dol_print_date($db->jdate($objp->eatby), 'day').'</td>';
```

Comment:
It's true that we must not use the $objp->eatby, but we must find the eatby value coming from the lot number (this is a property of couple lot / product.
But you must not find value with a request from product_batch or product_stock

This is a faster and better way to get the eatby

$lot=new Productlot($db);
$lot->fetch(0, $objp->pid, $objp->batch);

You msut just add pid that is id on product table into the sql request.